src/dashboards/components/AutoRefreshOverlay.tsx

```diff
@@ -0,0 +1,141 @@
+// Libraries
+import React, {FC, useContext, useCallback} from 'react'
+
+// Components
+import {
+  Overlay,
+  Grid,
+  Button,
+  ComponentColor,
+  SelectDropdown,
+} from '@influxdata/clockface'
+import {OverlayContext} from 'src/overlays/components/OverlayController'
+import AutoRefreshDropdown from 'src/shared/components/dropdown_auto_refresh/AutoRefreshDropdown'
+import TimeRangeDropdown from 'src/shared/components/DeleteDataForm/TimeRangeDropdown'
+
+// Types
+import {CustomTimeRange, AutoRefreshStatus} from 'src/types'
+
+// Context
+import AutoRefreshContextProvider, {
+  AutoRefreshContext,
+} from 'src/dashboards/components/RefreshContext'
+
+import './AutoRefresh.scss'
+
+const INACTIVITY_ARRAY = [...Array(25).keys()].map(num => num.toString())
```

Comment:
put a comment on this for what it does. cleverness is rarely rewarded as a code base lives on